src/lib/code-sample-preview.ts

```diff
@@ -128,19 +142,32 @@ export class CodeSamplePreviewElement extends LitElement {
 
   render() {
     return html`
-      <div id="toolbar">
+      <div id="toolbar" part="preview-toolbar">
+        <span id="location" part="preview-location"> ${this.location}</span>
         <mwc-icon-button
+          id="reload-button"
+          part="preview-reload-button"
           icon="refresh"
+          ?disabled=${!this.src}
           @click=${this._onReloadClick}
         ></mwc-icon-button>
-        <div id="location">${this.location}</div>
       </div>
-      ${this._iframeLoaded ? nothing : html`<slot></slot>`}
-      <iframe
-        src=${ifDefined(this.src)}
-        @load=${this._onIframeLoad}
-        ?hidden=${!this._iframeLoaded}
-      ></iframe>
+
+      <div id="content">
+        <mwc-linear-progress
```

Comment:
can you put this inside the toolbar instead, and position it absolute so it doesn't reserve space and shift the centering of the other items?

Reply:
also, can you make the background transparent so it doesn't show when not active?

Reply:
one more thing: I think it'll look better if the height of the progress bar is the same as the current tab line in the tab bar. I think it's 2px, but hopefully controlled by a css variable.

Reply:
I tried both. I think it looks better below the toolbar, because otherwise 1) it makes the toolbar contents look off-center when it's visible, and 2) this way the user can independently hide the toolbar and progress bar, since they have separate parts. It's already positioned absolutely above the contents of the iframe, and it's already transparent when disabled (that's just the default behavior or linear progress bar, it seems).

Reply:
It looks off to me that it doesn't line up with the current tab underline. I think (1) would be alleviated by the smaller height and if the background was transparent (not gray) even when active.

One way to keep it out of the toolbar would be to position it as `top: -2px` (or whatever the hight of the tab bar underline is).

Reply:
Ok, you convinced me, it looks much better in the toolbar once I shrink it to match the height of the tar-bar indicator. See new screenshot. The only downside here is that if you want to disable the toolbar, you have to also tweak the position of the loading indicator, but it's pretty straightforward:

```css
code-sample::part(preview-toolbar) {
  display: none;
}
code-sample::part(preview-loading-indicator) {
  top: 0
}
```

Reply:
Thanks for trying it all out!